pkg/handlers/host.go

```diff
@@ -107,9 +108,19 @@ func (h *HostHandlers) PatchHostByID(w http.ResponseWriter, req *http.Request) e
 	}
 	hostToDB, err := h.constructHostForDB(createHostRequest, req)
 	if err != nil {
-		return err
+		return api.WriteJSON(w, http.StatusInternalServerError, err.Error())
 	}
 	hostToDB.ID = id
+	hostGet, err := h.hostService.GetHostByID(hostToDB.ID)
+	if err != nil {
+		statusCode := http.StatusNotFound
+		return api.WriteJSON(w, statusCode, api.APIError{Error: http.StatusText(statusCode)})
+	}
+	if enums.IP.String() == createHostRequest.ValueType && len(hostGet.Domain) > 0 {
+		hostToDB.Domain = hostGet.Domain
+	} else if enums.Domain.String() == createHostRequest.ValueType && len(hostGet.IP) > 0 {
+		hostToDB.IP = hostGet.IP
+	}
 	host, err := h.hostService.PatchHostByID(hostToDB)
```

Comment:
Validation must take place in order to avoid letting the user input an invalid IP or Hostname. 

```suggestion
        // Get the value of the host, IP if it's an IP type, Hostname if it's a Domain/Subdomain
        if err := h.hostService.ValidateHost(hostValue); err != nil {
        // Handle the case when the host value (the target) is not valid...
        }
        // Patch the host in the DB if everything's ok
	host, err := h.hostService.PatchHostByID(hostToDB)
```

Reply:
Your are right, I have not considered it. Thanks for the suggestion